src/Email/EmailServiceProvider.php

```diff
@@ -31,6 +36,31 @@ public function register(Application $app)
                 $config['email']['mandrill_api_key'] = getenv('MANDRILL_API_KEY');
             }
         }
+        if (array_key_exists('mailgun_api_key', $config['email'])) {
+            $app['mailgun'] = new Mailgun($config['email']['mailgun_api_key']);
```

Comment:
Move this to MailgunSender itself, just like the Mandrill one.  As MailgunSender is the only thing which will use those things, there's no use in putting them in the application container.